TypeInjections/Tool/Tool.fs

```diff
@@ -21,77 +48,64 @@ module Tool =
+        let pathNew = $"{path}/New"
 
-            let commit1 = argvList.Item(1)
-            let commit2 = argvList.Item(2)
-
-            let path = argvList.Item(3)
-            let pathOld = $"{path}/Old"
-            let pathNew = $"{path}/New"
-
-            // lib2git needs an empty folder to store the repo clones
-            for a in [ pathOld; pathNew ] do
-                if Directory.Exists(a) then
+        // lib2git needs an empty folder to store the repo clones
+        for a in [ pathOld; pathNew ] do
+            if Directory.Exists(a) then
+                // if user lets us delete existing folder
+                if git.force then
                     Directory.Delete(a, true)
+                // else break program
+                else
+                    failwith
+                        $"{a} already exists. We need an empty folder to place checkout git repo. Choose different output folder or choose --force to automatically overwrite folder data"
 
-            checkoutCommit commit1 location pathOld
-            checkoutCommit commit2 location pathNew
+        checkoutCommit commit1 location pathOld
+        checkoutCommit commit2 location pathNew
 
+        Utils.log $"{pathOld}, {pathNew}, {path}/Output"
 
-        // -r [github repo url] [github commit hash] [github commit hash] [OutputPath]
-        elif flag.Contains("-r") then
 
-            Console.WriteLine("***** remote github")
+    let runLocal (local: LocalOptions) : unit =
 
-            if argvList.Length <> 5 then
-                failwith "usage: program -r GITHUB_URL OLD_COMMIT NEW_COMMIT OUTPUTFOLDER"
+        Utils.log "***** local project comparison"
+        Utils.log $"{local.oldFolder}, {local.newFolder}, {local.outputFolder}"
 
-            let url = argvList.Item(1)
 
-            let commit1 = argvList.Item(2)
-            let commit2 = argvList.Item(3)
+    [<EntryPoint>]
+    let main (argv: string array) =
 
-            let path = argvList.Item(4)
-            let pathOld = $"{path}/Old"
-            let pathNew = $"{path}/New"
+        Utils.log "***** Entering Tool.fs"
 
-            // lib2git requires an empty folder to place repos
-            for a in [ pathOld; pathNew ] do
-                if Directory.Exists(a) then
-                    Directory.Delete(a, true)
+        let result =
+            Parser.Default.ParseArguments<LocalOptions, GitOptions> argv
 
-            // grab repos of new and old projects
-            checkoutCommit commit1 url pathOld
-            checkoutCommit commit2 url pathNew
 
-        else
-            failwith "flag not found please try again"
+        match result with
+        | :? (Parsed<obj>) as command ->
+            match command.Value with
+            | :? LocalOptions as opts -> runLocal opts
+            | :? GitOptions as opts -> runGit opts
+            | _ -> 0 |> ignore
 
+        | :? (NotParsed<obj>) as value -> 0 |> ignore
+        | _ -> 0 |> ignore
```

Comment:
shouldn't all of these `0 |> ignore` print the help message and exit the program? Minor note, `0 |> ignore` could probably be simplified to `()`

Reply:
The `CommandLine.Parser` will automatically throw an error message, that is why I ignored the `NotParsed` sequence. I will simplify to `()`